From 723e3b8e79dca2453dadfa08eb2a97b7a6a5d9c8 Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Sun, 12 Feb 2017 15:57:24 +0100
Subject: [PATCH] CMake: Install pdb file for `Debug` and `RelWithDebInfo`

---
 CMakeLists.txt | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2b795fda1..f99a594e8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -684,6 +684,13 @@ if (NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL)
     FILES ${PROJECT_BINARY_DIR}/freetype-config-version.cmake
     DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/freetype
     COMPONENT headers)
+  if((MSVC_VERSION GREATER 1500) AND (NOT CMAKE_VERSION VERSION_LESS "3.1"))
+    install(FILES
+      $<TARGET_PDB_FILE:freetype>
+      DESTINATION lib
+      CONFIGURATIONS Debug RelWithDebInfo
+    )
+  endif()
 endif ()
 
 
-- 
2.14.3

