From b0515b6733994d46e5c17fcb99440db85a5df8ff Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Thu, 22 Mar 2018 21:35:38 +0100
Subject: [PATCH] CMake: When building a shared library export all symbols

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index cc9fc9545..fce28e8e2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -120,7 +120,7 @@ project(freetype)
 
 
 if (WIN32 AND NOT MINGW AND BUILD_SHARED_LIBS)
-  message(FATAL_ERROR "Building shared libraries on Windows needs MinGW")
+  set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
 endif ()
 
 # Disallow in-source builds
-- 
2.14.3

