From 291b8ba027a55f5f6123c3dd0bf63c9e39efb714 Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Thu, 22 Mar 2018 21:35:38 +0100
Subject: [PATCH] CMake: When building a shared library export all symbols

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f99a594e8..620f36f98 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -504,6 +504,7 @@ project(freetype)
   if (BUILD_SHARED_LIBS)
     target_compile_definitions(
       freetype PRIVATE DLL_EXPORT)
+    set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
   endif ()
 else ()
   # Windows handles exports with DLL_EXPORT defined above.
-- 
2.14.3

