From e135b5c81dcffab230d3e8050ae99e0a5a7e95c1 Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Sat, 24 Mar 2018 18:11:27 +0100
Subject: [PATCH] Remove shared library suffix on Windows

---
 CMakeLists.txt | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fce28e8e2..620f36f98 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -162,10 +162,6 @@ set(SHARED_LIBRARY_VERSION ${VERSION_MAJOR}.${VERSION_MINOR})
 
 # Set default version and debug suffix for libraries on MSVC
 if (MSVC)
-  foreach(CONFIG ${CMAKE_CONFIGURATION_TYPES})
-    string(TOUPPER ${CONFIG} CONFIG)
-    set(CMAKE_${CONFIG}_POSTFIX "${VERSION_MAJOR}${VERSION_MINOR}${VERSION_PATCH}MT")
-  endforeach()
   set(CMAKE_DEBUG_POSTFIX "${CMAKE_DEBUG_POSTFIX}d")
   add_definitions(-D_CRT_SECURE_NO_WARNINGS)
 endif ()
-- 
2.14.3

