From 200b90d4af63f9758a1185a6e733b536aec95ecf Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Mon, 21 May 2018 19:38:14 +0200
Subject: [PATCH] Don't export freetype cmake targets

---
 CMakeLists.txt | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 620f36f98..03d1714e4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -670,17 +670,11 @@ endif ()
 
   install(
     TARGETS freetype freetype-interface
-      EXPORT freetype-targets
       LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
       ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
       RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
       FRAMEWORK DESTINATION Library/Frameworks
       COMPONENT libraries)
-  install(
-    EXPORT freetype-targets
-      DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/freetype
-      FILE freetype-config.cmake
-      COMPONENT headers)
   install(
     FILES ${PROJECT_BINARY_DIR}/freetype-config-version.cmake
     DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/freetype
-- 
2.14.3

