diff --git a/CMakeLists.txt.orig b/CMakeLists.txt
index a7a173d..caccbdf 100644
--- a/CMakeLists.txt.orig
+++ b/CMakeLists.txt
@@ -109,6 +109,7 @@ if (MSVC)
     set(CMAKE_${CONFIG}_POSTFIX "${VERSION_MAJOR}${VERSION_MINOR}${VERSION_PATCH}MT")
   endforeach()
   set(CMAKE_DEBUG_POSTFIX "${CMAKE_DEBUG_POSTFIX}d")
+  add_definitions(-D_CRT_SECURE_NO_WARNINGS)
 endif ()
 
 # Compiler definitions for building the library
