diff --git a/CMakeLists.txt b/CMakeLists.txt
index f989049..a6d6c1b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -110,3 +110,10 @@ install(TARGETS ogg RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} LIBRARY DESTINAT
 if(NOT WIN32 OR CYGWIN)
     install(FILES ogg.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
 endif()
+if(BUILD_SHARED_LIBS AND (MSVC_VERSION GREATER 1500) AND (NOT CMAKE_VERSION VERSION_LESS "3.1"))
+    install(
+        FILES $<TARGET_PDB_FILE:ogg>
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        CONFIGURATIONS Debug RelWithDebInfo
+    )
+endif()
