--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -37,6 +37,7 @@ OPTION(ALSOFT_EXAMPLES  "Build and install example programs"  ON)
 
 OPTION(ALSOFT_CONFIG "Install alsoft.conf sample configuration file" ON)
 OPTION(ALSOFT_HRTF_DEFS "Install HRTF definition files" ON)
+OPTION(ALSOFT_PKGCONFIG "Install pkg-config file" ON)
 
 
 IF(WIN32)
@@ -1021,10 +1022,13 @@ SET(PACKAGE_VERSION "${LIB_VERSION}")
 CONFIGURE_FILE(
     "${OpenAL_SOURCE_DIR}/config.h.in"
     "${OpenAL_BINARY_DIR}/config.h")
-CONFIGURE_FILE(
-    "${OpenAL_SOURCE_DIR}/openal.pc.in"
-    "${OpenAL_BINARY_DIR}/openal.pc"
-    @ONLY)
+
+IF(ALSOFT_PKGCONFIG)
+    CONFIGURE_FILE(
+        "${OpenAL_SOURCE_DIR}/openal.pc.in"
+        "${OpenAL_BINARY_DIR}/openal.pc"
+        @ONLY)
+ENDIF()
 
 # Build a common library with reusable helpers
 ADD_LIBRARY(common STATIC ${COMMON_OBJS})
@@ -1107,9 +1111,15 @@ INSTALL(FILES include/AL/al.h
               include/AL/efx-presets.h
         DESTINATION include/AL
 )
-INSTALL(FILES "${OpenAL_BINARY_DIR}/openal.pc"
-        DESTINATION "lib${LIB_SUFFIX}/pkgconfig")
-
+IF(ALSOFT_PKGCONFIG)
+    INSTALL(FILES "${OpenAL_BINARY_DIR}/openal.pc"
+            DESTINATION "lib${LIB_SUFFIX}/pkgconfig")
+ENDIF()
+IF(LIBTYPE STREQUAL "SHARED")
+    INSTALL(FILES "${CMAKE_CURRENT_BINARY_DIR}/${LIBNAME}.pdb"
+            DESTINATION "lib${LIB_SUFFIX}"
+            CONFIGURATIONS Debug RelWithDebInfo)
+ENDIF()
 
 MESSAGE(STATUS "")
 MESSAGE(STATUS "Building OpenAL with support for the following backends:")
