index aa37f67..8425360 100644
--- a/cmake/libpython/CMakeLists.txt
+++ b/cmake/libpython/CMakeLists.txt
@@ -196,6 +196,9 @@ elseif(WIN32)
         SOURCE ${SRC_DIR}/PC/dl_nt.c
         PROPERTY COMPILE_DEFINITIONS Py_ENABLE_SHARED
     )
+    if(MSVC)
+        add_compile_options(/wd4101 /wd4018 /wd4244 /wd4996)
+    endif()
 endif(UNIX)
 
 # List of builtin extensions
--- a/cmake/pgen/CMakeLists.txt
+++ b/cmake/pgen/CMakeLists.txt
@@ -22,5 +22,9 @@
     ${SRC_DIR}/Python/mysnprintf.c
     ${SRC_DIR}/Python/pyctype.c
 )
+
+if(MSVC)
+    add_compile_options(/wd4018)
+endif()
 
 add_executable(pgen ${PGEN_SOURCES})
