From 66637bdb6014e10b58499f13a62870ef533551f6 Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Wed, 9 Mar 2016 16:24:16 +0100
Subject: [PATCH] CMake: Remove vorbisenc target

---
 lib/CMakeLists.txt | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index d500287..8d67517 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -75,19 +75,16 @@ endif()
 
 if (NOT BUILD_FRAMEWORK)
     add_library(vorbis ${VORBIS_HEADERS} ${VORBIS_SOURCES})
-    add_library(vorbisenc ${VORBISENC_SOURCES})
     add_library(vorbisfile ${VORBISFILE_SOURCES})
 
     set_target_properties(
-        vorbis vorbisenc vorbisfile PROPERTIES
+        vorbis vorbisfile PROPERTIES
         IMPORT_PREFIX "lib"
         PREFIX "lib"
     )
 
     get_version_info(VORBIS_VERSION_INFO "V_LIB_CURRENT" "V_LIB_AGE" "V_LIB_REVISION")
     set_target_properties(vorbis PROPERTIES SOVERSION ${VORBIS_VERSION_INFO})
-    get_version_info(VORBISENC_VERSION_INFO "VE_LIB_CURRENT" "VE_LIB_AGE" "VE_LIB_REVISION")
-    set_target_properties(vorbisenc PROPERTIES SOVERSION ${VORBISENC_VERSION_INFO})
     get_version_info(VORBISFILE_VERSION_INFO "VF_LIB_CURRENT" "VF_LIB_AGE" "VF_LIB_REVISION")
     set_target_properties(vorbisfile PROPERTIES SOVERSION ${VORBISFILE_VERSION_INFO})
 
@@ -98,13 +95,6 @@ if (NOT BUILD_FRAMEWORK)
        PRIVATE
             ${CMAKE_CURRENT_SOURCE_DIR}
     )
-    target_include_directories(vorbisenc
-        PUBLIC
-            $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
-            $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
-        PRIVATE
-            ${CMAKE_CURRENT_SOURCE_DIR}
-    )
     target_include_directories(vorbisfile
         PUBLIC
             $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
@@ -115,12 +105,11 @@ if (NOT BUILD_FRAMEWORK)
         PUBLIC Ogg::ogg
         PRIVATE $<$<BOOL:${HAVE_LIBM}>:m>
     )
-    target_link_libraries(vorbisenc PUBLIC vorbis)
     target_link_libraries(vorbisfile PUBLIC vorbis)
 
     install(FILES ${VORBIS_PUBLIC_HEADERS} DESTINATION ${CMAKE_INSTALL_FULL_INCLUDEDIR}/vorbis)
 
-    install(TARGETS vorbis vorbisenc vorbisfile
+    install(TARGETS vorbis vorbisfile
         EXPORT VorbisTargets
         RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
         LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
@@ -131,7 +120,6 @@ if (NOT BUILD_FRAMEWORK)
         install(
             FILES
                 $<TARGET_PDB_FILE:vorbis>
-                $<TARGET_PDB_FILE:vorbisenc>
                 $<TARGET_PDB_FILE:vorbisfile>
             DESTINATION ${CMAKE_INSTALL_LIBDIR}
             CONFIGURATIONS Debug RelWithDebInfo
-- 
2.7.4

