From 1afdb58263f97ccb899e4211aaadc9cbbf9b5474 Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Thu, 31 Mar 2016 17:03:28 +0200
Subject: [PATCH] CMake: Use install command for both framework and
 non-framework

---
 lib/CMakeLists.txt | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 16fa575..6f64087 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -110,7 +110,7 @@ if (NOT BUILD_FRAMEWORK)
 
     install(FILES ${VORBIS_PUBLIC_HEADERS} DESTINATION ${CMAKE_INSTALL_FULL_INCLUDEDIR}/vorbis)
 
-    install(TARGETS vorbis vorbisfile
+    install(TARGETS vorbisfile
         EXPORT VorbisTargets
         RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
         LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
@@ -115,3 +114,10 @@ else()
     )
     target_link_libraries(vorbis ${OGG_LIBRARIES})
 endif()
+
+install(TARGETS vorbis
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    FRAMEWORK DESTINATION ${CMAKE_INSTALL_PREFIX}
+)
-- 
2.7.4

