From 0f4c4afa1ee03d361a47e3662e37933a34a2155d Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Thu, 6 Oct 2016 23:31:52 +0200
Subject: [PATCH] Don't install glew CMake export targets

---
 build/cmake/CMakeLists.txt | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/build/cmake/CMakeLists.txt b/build/cmake/CMakeLists.txt
index 7920d57..2a41cb7 100644
--- a/build/cmake/CMakeLists.txt
+++ b/build/cmake/CMakeLists.txt
@@ -242,15 +242,6 @@ install (FILES
     ${GLEW_DIR}/include/GL/glxew.h
     DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/GL)
 
-if(MAYBE_EXPORT)
-  install(EXPORT glew-targets DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/glew
-    NAMESPACE GLEW::)
-  install(FILES
-      ${CMAKE_CURRENT_SOURCE_DIR}/glew-config.cmake
-      ${CMAKE_CURRENT_SOURCE_DIR}/CopyImportedTargetProperties.cmake
-    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/glew)
-endif()
-
 if(NOT TARGET uninstall)
   configure_file(
       ${CMAKE_CURRENT_SOURCE_DIR}/cmake_uninstall.cmake.in
-- 
2.14.3

