From 02f63a0a09c449697ee3a9156467b6b8ae7bb516 Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Tue, 8 Mar 2016 10:56:04 +0100
Subject: [PATCH] CMake: Install pdb file for `Debug` and `RelWithDebInfo`

---
 CMakeLists.txt | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 385697a..bf4d7ed 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -116,3 +116,10 @@ install(TARGETS ogg
 if(NOT WIN32 OR CYGWIN)
     install(FILES ogg.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
 endif()
+if(BUILD_SHARED_LIBS AND (MSVC_VERSION GREATER 1500) AND (NOT CMAKE_VERSION VERSION_LESS "3.1"))
+    install(
+        FILES $<TARGET_PDB_FILE:ogg>
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        CONFIGURATIONS Debug RelWithDebInfo
+    )
+endif()
-- 
2.7.4

