From 9962c5314c67bb0d2e1ce836e7a698ba77c1c3ae Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Wed, 28 Sep 2016 16:13:30 +0200
Subject: [PATCH] Don't install pkg-config files when building a MacOSX
 framework

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index bf4d7ed..e1b67f2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -113,7 +113,7 @@ install(TARGETS ogg
     FRAMEWORK DESTINATION ${CMAKE_INSTALL_PREFIX}
     PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/ogg
 )
-if(NOT WIN32 OR CYGWIN)
+if(NOT BUILD_FRAMEWORK AND (NOT WIN32 OR CYGWIN))
     install(FILES ogg.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
 endif()
 if(BUILD_SHARED_LIBS AND (MSVC_VERSION GREATER 1500) AND (NOT CMAKE_VERSION VERSION_LESS "3.1"))
-- 
2.7.4

