From 11d4122cb675e87c2d56ddc9eee89fcd0504b532 Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Wed, 14 Sep 2016 10:41:37 +0200
Subject: [PATCH] Install pdb file for Debug and RelWithDebInfo configurations

---
 CMakeLists.txt | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 48c6fa287..de67078c7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -879,6 +879,12 @@ if(NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL )
       CREATE_SYMLINK(libpng${CMAKE_SHARED_LIBRARY_SUFFIX} TARGET png)
       install(FILES $<TARGET_LINKER_FILE_DIR:png>/libpng${CMAKE_SHARED_LIBRARY_SUFFIX} DESTINATION ${CMAKE_INSTALL_LIBDIR})
     endif(NOT WIN32)
+
+    if((MSVC_VERSION GREATER 1500) AND (NOT CMAKE_VERSION VERSION_LESS "3.1") AND PNG_SHARED AND NOT SKIP_INSTALL_DEBUG_SYMBOLS)
+      install(FILES $<TARGET_PDB_FILE:png>
+        DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+        CONFIGURATIONS Debug RelWithDebInfo)
+    endif()
   endif(PNG_SHARED)
 
   if(PNG_STATIC)
-- 
2.14.3

