From e6da99dc319bb9e066184de285928eadc370b91b Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Mon, 6 Feb 2017 21:40:42 +0100
Subject: [boost] Remove version suffix from include directory

---
 boostcpp.jam | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/boostcpp.jam b/boostcpp.jam
index 85a4cb323..6fe87d447 100644
--- a/boostcpp.jam
+++ b/boostcpp.jam
@@ -267,15 +267,7 @@ rule declare_install_and_stage_proper_targets ( libraries * : headers * )
 
     install-requirements = <install-source-root>$(BOOST_ROOT)/boost ;
 
-    if $(layout-versioned)
-    {
-        install-requirements +=
-            <install-header-subdir>boost-$(BOOST_VERSION_TAG)/boost ;
-    }
-    else
-    {
-        install-requirements += <install-header-subdir>boost ;
-    }
+    install-requirements += <install-header-subdir>boost ;
 
     if [ os.name ] = NT
     {
-- 
2.14.3

