From 5e1665a7c739103359e8ac63e43f763a3d24929f Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Mon, 6 Feb 2017 21:40:42 +0100
Subject: [PATCH] Remove version suffix from include directory

---
 boostcpp.jam | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/boostcpp.jam b/boostcpp.jam
index f45d12e..29e80c4 100644
--- a/boostcpp.jam
+++ b/boostcpp.jam
@@ -254,15 +254,7 @@ rule declare_install_and_stage_proper_targets ( libraries * : headers * )
 
     install-requirements = <install-source-root>$(BOOST_ROOT)/boost ;
 
-    if $(layout-versioned)
-    {
-        install-requirements +=
-            <install-header-subdir>boost-$(BOOST_VERSION_TAG)/boost ;
-    }
-    else
-    {
-        install-requirements += <install-header-subdir>boost ;
-    }
+    install-requirements += <install-header-subdir>boost ;
 
     if [ os.name ] = NT
     {
-- 
2.9.3

