From 6fdf8a933db4ca3be83190f54056db9a15f67582 Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Thu, 10 Mar 2016 09:31:51 +0100
Subject: [PATCH 1/2] Install pdb file for `Debug` and `RelWithDebInfo`

---
 CMakeLists.txt | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0c0247c..54f03f5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -225,6 +225,13 @@ endif()
 if(NOT SKIP_INSTALL_FILES AND NOT SKIP_INSTALL_ALL )
     install(FILES ${ZLIB_PC} DESTINATION "${INSTALL_PKGCONFIG_DIR}")
 endif()
+if((MSVC_VERSION GREATER 1500) AND (NOT CMAKE_VERSION VERSION_LESS "3.1") AND NOT SKIP_INSTALL_DEBUG_INFO AND NOT SKIP_INSTALL_ALL)
+    install(
+        FILES $<TARGET_PDB_FILE:zlib>
+        DESTINATION "${INSTALL_LIB_DIR}"
+        CONFIGURATIONS Debug RelWithDebInfo
+    )
+endif()
 
 #============================================================================
 # Example binaries
-- 
2.5.0

