From 2712ebde5d80728ced03ac14ef1121cfead9aab5 Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Fri, 1 Apr 2016 17:21:45 +0200
Subject: [PATCH] CMake: Install pdb file for Debug and RelWithDebInfo
 configuration

---
 CMakeLists.txt | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8f1e828d1..aa17932b6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1947,6 +1947,13 @@ if(NOT (WINDOWS OR CYGWIN))
   install(FILES "${SDL2_SOURCE_DIR}/sdl2.m4" DESTINATION "${CMAKE_INSTALL_FULL_DATAROOTDIR}/aclocal")
 endif()
 
+if(SDL_SHARED AND (MSVC_VERSION GREATER 1500) AND (NOT CMAKE_VERSION VERSION_LESS "3.1"))
+  install(FILES
+    $<TARGET_PDB_FILE:SDL2>
+    DESTINATION "lib"
+    CONFIGURATIONS Debug RelWithDebInfo)
+endif()
+
 ##### Uninstall target #####
 
 if(NOT TARGET uninstall)
-- 
2.14.3

