From b8b84a04806f077a341a6014ae71ce0ac9d182ad Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Fri, 1 Apr 2016 17:23:05 +0200
Subject: [PATCH] Never build and install SDL2main library

---
 CMakeLists.txt | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f951d0e..81257e0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2747,17 +2747,17 @@ endif()
 # Ensure that the extra cflags are used at compile time
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${EXTRA_CFLAGS}")
 
-# Always build SDLmain
-if(NOT WINDOWS_STORE)
-  add_library(SDL2main STATIC ${SDLMAIN_SOURCES})
-  # alias target for in-tree builds
-  add_library(SDL2::SDL2main ALIAS SDL2main)
-  target_include_directories(SDL2main BEFORE PRIVATE "${SDL2_BINARY_DIR}/include")
-  target_include_directories(SDL2main PUBLIC "$<BUILD_INTERFACE:${SDL2_SOURCE_DIR}/include>" $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}> $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/SDL2>)
-  if (NOT ANDROID)
-    set_target_properties(SDL2main PROPERTIES DEBUG_POSTFIX "${SDL_CMAKE_DEBUG_POSTFIX}")
-  endif()
-endif()
+# Never build SDLmain
+#if(NOT WINDOWS_STORE)
+#  add_library(SDL2main STATIC ${SDLMAIN_SOURCES})
+#  # alias target for in-tree builds
+#  add_library(SDL2::SDL2main ALIAS SDL2main)
+#  target_include_directories(SDL2main BEFORE PRIVATE "${SDL2_BINARY_DIR}/include")
+#  target_include_directories(SDL2main PUBLIC "$<BUILD_INTERFACE:${SDL2_SOURCE_DIR}/include>" $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}> $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/SDL2>)
+#  if (NOT ANDROID)
+#    set_target_properties(SDL2main PROPERTIES DEBUG_POSTFIX "${SDL_CMAKE_DEBUG_POSTFIX}")
+#  endif()
+#endif()
 
 if(ANDROID)
   target_include_directories(sdl-build-options INTERFACE "${ANDROID_NDK}/sources/android/cpufeatures")
@@ -2858,12 +2858,12 @@ if(SDL_SHARED)
     RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}")
 endif()
 
-if(NOT WINDOWS_STORE)
-  install(TARGETS SDL2main EXPORT SDL2mainTargets
-    LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-    ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}")
-endif()
+#if(NOT WINDOWS_STORE)
+#  install(TARGETS SDL2main EXPORT SDL2mainTargets
+#    LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+#    ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+#    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}")
+#endif()
 
 if(SDL_STATIC)
   install(TARGETS SDL2-static EXPORT SDL2staticTargets
@@ -2893,13 +2893,13 @@ if(SDL_SHARED)
   )
 endif()
 
-if(NOT WINDOWS_STORE)
-  install(EXPORT SDL2mainTargets
-    FILE SDL2mainTargets.cmake
-    NAMESPACE SDL2::
-    DESTINATION ${PKG_PREFIX}
-  )
-endif()
+#if(NOT WINDOWS_STORE)
+#  install(EXPORT SDL2mainTargets
+#    FILE SDL2mainTargets.cmake
+#    NAMESPACE SDL2::
+#    DESTINATION ${PKG_PREFIX}
+#  )
+#endif()
 
 if(SDL_STATIC)
   install(EXPORT SDL2staticTargets
-- 
2.14.3

