From 43524a3266a76d091e41c4aca86e009fd06bae9d Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Fri, 1 Apr 2016 18:10:19 +0200
Subject: [PATCH] CMake: Add MacOSX Framework support

---
 CMakeLists.txt | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ec2cdd4..c3185d3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -419,6 +419,9 @@ option_string(BACKGROUNDING_SIGNAL "number to use for magic backgrounding signal
 set_option(SDL_HIDAPI_JOYSTICK     "Use HIDAPI for low level joystick drivers" ON)
 set_option(SDL_VIRTUAL_JOYSTICK    "Enable the virtual-joystick driver" ON)
 set_option(SDL_ASAN                "Use AddressSanitizer to detect memory errors" OFF)
+if(APPLE)
+  set_option(BUILD_FRAMEWORK   "Build Framework bundle for OSX" OFF)
+endif()
 
 set(SDL_SHARED ${SDL_SHARED_ENABLED_BY_DEFAULT} CACHE BOOL "Build a shared version of the library")
 set(SDL_STATIC ${SDL_STATIC_ENABLED_BY_DEFAULT} CACHE BOOL "Build a static version of the library")
@@ -2731,6 +2734,16 @@ if(SDL_SHARED)
     set_target_properties(SDL2 PROPERTIES STATIC_LIBRARY_FLAGS "/NODEFAULTLIB")
   endif()
   set(_INSTALL_LIBS "SDL2" ${_INSTALL_LIBS})
+  if(APPLE AND BUILD_FRAMEWORK)
+    set_target_properties(SDL2 PROPERTIES
+      FRAMEWORK TRUE
+      FRAMEWORK_VERSION ${SDL_VERSION}
+      MACOSX_FRAMEWORK_IDENTIFIER org.libsdl.SDL2
+      MACOSX_FRAMEWORK_SHORT_VERSION_STRING ${SDL_VERSION}
+      MACOSX_FRAMEWORK_BUNDLE_VERSION ${SDL_VERSION}
+      MACOSX_RPATH ON
+      OUTPUT_NAME "SDL2")
+  endif()
   target_link_libraries(SDL2 PRIVATE ${EXTRA_LIBS} ${EXTRA_LDFLAGS})
   target_include_directories(SDL2 BEFORE PRIVATE "${SDL2_BINARY_DIR}/include")
   target_include_directories(SDL2 PUBLIC "$<BUILD_INTERFACE:${SDL2_SOURCE_DIR}/include>;$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>;$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/SDL2>")
-- 
2.14.3

