From 43524a3266a76d091e41c4aca86e009fd06bae9d Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Fri, 1 Apr 2016 18:10:19 +0200
Subject: [PATCH] CMake: Add MacOSX Framework support

---
 CMakeLists.txt | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 707eb5c..af93a52 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -505,6 +505,9 @@ dep_option(SDL_HIDAPI_LIBUSB       "Use libusb for low level joystick drivers" O
 dep_option(SDL_HIDAPI_JOYSTICK     "Use HIDAPI for low level joystick drivers" ON SDL_HIDAPI OFF)
 dep_option(SDL_VIRTUAL_JOYSTICK    "Enable the virtual-joystick driver" ON SDL_HIDAPI OFF)
 set_option(SDL_ASAN                "Use AddressSanitizer to detect memory errors" OFF)
+if(APPLE)
+  set_option(BUILD_FRAMEWORK   "Build Framework bundle for OSX" OFF)
+endif()
 
 option(SDL_SHARED "Build a shared version of the library" ${SDL_SHARED_ENABLED_BY_DEFAULT})
 option(SDL_STATIC "Build a static version of the library" ${SDL_STATIC_ENABLED_BY_DEFAULT})
@@ -3097,6 +3100,16 @@ if(SDL_SHARED)
     endif()
     set_target_properties(SDL2 PROPERTIES STATIC_LIBRARY_FLAGS "/NODEFAULTLIB")
   endif()
+  if(APPLE AND BUILD_FRAMEWORK)
+    set_target_properties(SDL2 PROPERTIES
+      FRAMEWORK TRUE
+      FRAMEWORK_VERSION ${SDL_VERSION}
+      MACOSX_FRAMEWORK_IDENTIFIER org.libsdl.SDL2
+      MACOSX_FRAMEWORK_SHORT_VERSION_STRING ${SDL_VERSION}
+      MACOSX_FRAMEWORK_BUNDLE_VERSION ${SDL_VERSION}
+      MACOSX_RPATH ON
+      OUTPUT_NAME "SDL2")
+  endif()
   # FIXME: if CMAKE_VERSION >= 3.13, use target_link_options for EXTRA_LDFLAGS
   target_link_libraries(SDL2 PRIVATE ${EXTRA_LIBS} ${EXTRA_LDFLAGS} ${EXTRA_LDFLAGS_BUILD})
   target_include_directories(SDL2 PUBLIC
@@ -3288,7 +3301,7 @@ if(NOT SDL2_DISABLE_INSTALL)
     set(SOPOSTFIX "")
   endif()
 
-  if(NOT (WINDOWS OR CYGWIN) OR MINGW)
+  if((NOT (WINDOWS OR CYGWIN) OR MINGW) AND (NOT BUILD_FRAMEWORK))
     if(SDL_SHARED)
       set(SOEXT ${CMAKE_SHARED_LIBRARY_SUFFIX}) # ".so", ".dylib", etc.
       get_target_property(SONAME SDL2 OUTPUT_NAME)
-- 
2.14.3

