From 5fd4d7b5908d51531e5a83d34621f41d10270959 Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Fri, 1 Apr 2016 18:58:04 +0200
Subject: [PATCH] CMake: Set framework install directory

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b37c4ab1e..ef2aa64d1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1793,6 +1793,7 @@ install(TARGETS ${_INSTALL_LIBS} EXPORT SDL2Targets
   LIBRARY DESTINATION "lib${LIB_SUFFIX}"
   ARCHIVE DESTINATION "lib${LIB_SUFFIX}"
   RUNTIME DESTINATION bin
+  FRAMEWORK DESTINATION ${CMAKE_INSTALL_PREFIX}
   PUBLIC_HEADER DESTINATION include/SDL2)
 
 ##### Export files #####
-- 
2.14.3

