From d71b9e6637e822ea334dfd64248f2f5ab0d79e89 Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Fri, 1 Apr 2016 18:10:19 +0200
Subject: [PATCH] CMake: Add MacOSX Framework support

---
 CMakeLists.txt | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 134da54..a0434ca 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -293,6 +293,9 @@ set_option(VIDEO_COCOA         "Use Cocoa video driver" ${APPLE})
 set_option(DIRECTX             "Use DirectX for Windows audio/video" ${WINDOWS})
 set_option(RENDER_D3D          "Enable the Direct3D render driver" ${WINDOWS})
 set_option(VIDEO_VIVANTE       "Use Vivante EGL video driver" ${UNIX_SYS})
+if(APPLE)
+  set_option(BUILD_FRAMEWORK   "Build Framework bundle for OSX" OFF)
+endif()
 
 # TODO: We should (should we?) respect cmake's ${BUILD_SHARED_LIBS} flag here
 # The options below are for compatibility to configure's default behaviour.
@@ -1482,6 +1485,17 @@ if(SDL_SHARED)
   endif()
   set(_INSTALL_LIBS "SDL2" ${_INSTALL_LIBS})
   target_link_libraries(SDL2 ${EXTRA_LIBS} ${EXTRA_LDFLAGS})
+  if(APPLE AND BUILD_FRAMEWORK)
+    set_target_properties(SDL2 PROPERTIES
+      FRAMEWORK TRUE
+      FRAMEWORK_VERSION ${SDL_VERSION}
+      MACOSX_FRAMEWORK_IDENTIFIER org.libsdl.SDL2
+      MACOSX_FRAMEWORK_SHORT_VERSION_STRING ${SDL_VERSION}
+      MACOSX_FRAMEWORK_BUNDLE_VERSION ${SDL_VERSION}
+      XCODE_ATTRIBUTE_INSTALL_PATH "@rpath"
+      OUTPUT_NAME "SDL2")
+  endif()
+ target_link_libraries(SDL2 ${EXTRA_LIBS} ${EXTRA_LDFLAGS})
 endif()
 
 if(SDL_STATIC)
@@ -1508,7 +1522,7 @@ install(TARGETS ${_INSTALL_LIBS}
   RUNTIME DESTINATION bin
   PUBLIC_HEADER DESTINATION include/SDL2)
 
-if(NOT WINDOWS OR CYGWIN)
+if((NOT WINDOWS OR CYGWIN) AND (NOT BUILD_FRAMEWORK))
   if(SDL_SHARED)
     install(CODE "
       execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink
-- 
2.7.4

