From fbd7a965988f842ae8f2c43f38de726afd649314 Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Sat, 2 Apr 2016 12:18:44 +0200
Subject: [PATCH] Install pdb file for Debug and RelWithDebInfo configurations

---
 CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e5b836cd..a151fd3b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1355,6 +1355,11 @@ IF(ALSOFT_INSTALL)
         )
     ENDIF()
 ENDIF()
+IF(LIBTYPE STREQUAL "SHARED" AND (MSVC_VERSION GREATER 1500) AND (NOT CMAKE_VERSION VERSION_LESS "3.1"))
+    INSTALL(FILES $<TARGET_PDB_FILE:OpenAL>
+            DESTINATION "lib${LIB_SUFFIX}"
+            CONFIGURATIONS Debug RelWithDebInfo)
+ENDIF()
 
 
 if(HAS_ROUTER)
-- 
2.14.3

