From 98bcffd0161c5fe0c43e238c912918551de4b3f3 Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Mon, 21 Mar 2016 20:39:55 +0100
Subject: [PATCH] Change pathes to match current SDK layout

---
 CMakeLists.txt | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 91a5ce3..f364089 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -321,10 +321,7 @@ if(USE_LIB64)
     set_property(GLOBAL PROPERTY FIND_LIBRARY_USE_LIB64_PATHS ON)
     set(LIBDIR "lib64")
 endif()
-set(PYTHONHOME "${LIBDIR}")
-if(UNIX)
-    set(PYTHONHOME "${PYTHONHOME}/${LIBPYTHON}")
-endif()
+set(PYTHONHOME "${LIBDIR}/${LIBPYTHON}")
 
 # Install tree directory
 set(BIN_INSTALL_DIR bin)                         # Contains the python executable
@@ -346,7 +343,7 @@ if(INSTALL_WINDOWS_TRADITIONAL)
 endif()
 set(INCLUDE_INSTALL_DIR include/python${LD_VERSION})
 if(MSVC)
-    set(INCLUDE_INSTALL_DIR include)
+    set(INCLUDE_INSTALL_DIR include/python${PY_VERSION_MAJOR}.${PY_VERSION_MINOR})
 endif()
 # Build tree directory
 set(BIN_BUILD_DIR ${PROJECT_BINARY_DIR}/bin)
@@ -354,7 +351,7 @@ set(CONFIG_BUILD_DIR ${PROJECT_BINARY_DIR}/${CONFIG_INSTALL_DIR})
 set(EXTENSION_BUILD_DIR ${PROJECT_BINARY_DIR}/${PYTHONHOME}/lib-dynload)
 set(INCLUDE_BUILD_DIR ${SRC_DIR}/Include)
 
-set(ARCHIVEDIR "libs")      # Contains the static (*.a) and import libraries (*.lib)
+set(ARCHIVEDIR "lib")      # Contains the static (*.a) and import libraries (*.lib)
 
 # Directories specific to 'libpython'
 set(LIBPYTHON_LIBDIR ${LIBDIR})
-- 
2.14.3

