From 30b9a7e39404dcca8f771be2c218864b4293708a Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Tue, 13 Sep 2016 17:10:25 +0200
Subject: [PATCH] Include backported Visual Studio 2015 patch into build

---
 cmake/libpython/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/cmake/libpython/CMakeLists.txt b/cmake/libpython/CMakeLists.txt
index a33084f..8c6b80b 100644
--- a/cmake/libpython/CMakeLists.txt
+++ b/cmake/libpython/CMakeLists.txt
@@ -99,6 +99,9 @@ set(OBJECT3_SOURCES
     ${SRC_DIR}/Objects/odictobject.c
 )
 if(MSVC)
+    list(APPEND OBJECT2_SOURCES
+        ${SRC_DIR}/PC/invalid_parameter_handler.c
+    )
     list(APPEND OBJECT3_SOURCES
         ${SRC_DIR}/PC/invalid_parameter_handler.c
     )
-- 
2.14.3

