From f047845c5c8fcc968ebd38a22cc250d8479c82bb Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Tue, 13 Sep 2016 17:10:25 +0200
Subject: [PATCH] Include backported Visual Studio 2015 patch into build

---
 cmake/libpython/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/cmake/libpython/CMakeLists.txt b/cmake/libpython/CMakeLists.txt
index db93b24..4c53c5b 100644
--- a/cmake/libpython/CMakeLists.txt
+++ b/cmake/libpython/CMakeLists.txt
@@ -92,6 +92,9 @@ set(OBJECT3_SOURCES
     ${SRC_DIR}/Objects/odictobject.c
 )
 if(MSVC)
+    list(APPEND OBJECT2_SOURCES
+        ${SRC_DIR}/PC/invalid_parameter_handler.c
+    )
     list(APPEND OBJECT3_SOURCES
         ${SRC_DIR}/PC/invalid_parameter_handler.c
     )
