From f7a1a85885ec6faadf33bc3d3d3acff9b27c2d4f Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Tue, 20 Sep 2016 17:21:09 +0200
Subject: [PATCH] Install platform library files on MacOSX

---
 cmake/lib/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/cmake/lib/CMakeLists.txt b/cmake/lib/CMakeLists.txt
index f3a7c96..a6eef10 100644
--- a/cmake/lib/CMakeLists.txt
+++ b/cmake/lib/CMakeLists.txt
@@ -7,6 +7,10 @@ if(UNIX)
     set(plat_subdir "plat-linux2")
 endif()
 
+if(APPLE)
+    set(plat_subdir "plat-darwin")
+endif()
+
 foreach(file ${libfiles})
     # Don't install files for other platforms
     string(REGEX MATCH "^plat-" is_platform_file "${file}")
-- 
2.14.3

