From ad1e856852295e9123f77db27d88512218fbe4f1 Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Tue, 15 Nov 2016 10:57:52 +0100
Subject: [PATCH 12/14] Set INSTALL_NAME_DIR only for non-frameworks on OSX

---
 cmake/libpython/CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/cmake/libpython/CMakeLists.txt b/cmake/libpython/CMakeLists.txt
index f3467e1..ea6da58 100644
--- a/cmake/libpython/CMakeLists.txt
+++ b/cmake/libpython/CMakeLists.txt
@@ -568,7 +568,6 @@ function(add_libpython name type install component)
         OUTPUT_NAME ${LIBPYTHON}${ABIFLAGS}
         LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/${LIBPYTHON_LIBDIR}
         RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/${LIBPYTHON_LIBDIR}
-        INSTALL_NAME_DIR ${CMAKE_INSTALL_PREFIX}/${LIBPYTHON_LIBDIR}
     )
     set_target_properties(${name} PROPERTIES
         POSITION_INDEPENDENT_CODE ON
@@ -587,6 +586,10 @@ function(add_libpython name type install component)
             MACOSX_RPATH ON
             OUTPUT_NAME ${FRAMEWORK_NAME}
         )
+    else()
+        set_target_properties(${name} PROPERTIES
+            INSTALL_NAME_DIR ${CMAKE_INSTALL_PREFIX}/${LIBPYTHON_LIBDIR}
+        )
     endif()
 
     if(install)
-- 
2.35.1

