From 6be4a26c8ce4a93314865974304348d27b49525f Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Thu, 23 Mar 2017 22:51:06 +0100
Subject: [PATCH] Set python exe RPATH to search for libpython in Frameworks

---
 cmake/python/CMakeLists.txt | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/cmake/python/CMakeLists.txt b/cmake/python/CMakeLists.txt
index c7f01d6..d6137e5 100644
--- a/cmake/python/CMakeLists.txt
+++ b/cmake/python/CMakeLists.txt
@@ -27,6 +27,12 @@ if(MSVC)
     set_target_properties(python PROPERTIES LINK_FLAGS /STACK:2000000)
 endif()
 
+if(BUILD_FRAMEWORK)
+    set_target_properties(python PROPERTIES
+        INSTALL_RPATH "@loader_path/../../../.."
+    )
+endif()
+
 # Export target
 set_property(GLOBAL APPEND PROPERTY PYTHON_TARGETS python)
 
-- 
2.14.3

